src/client/clientmedia.cpp

```diff
@@ -195,6 +200,13 @@ void ClientMediaDownloader::initialStep(Client *client)
 			filestatus->received = true;
 			m_uncached_count--;
 		}
+
+		u64 cur_time = porting::getTimeMs();
+		u64 dtime = porting::getDeltaMs(last_time, cur_time);
+		if (dtime >= chunk_time_ms) {
+			client->drawLoadScreen(loading_text, dtime / 1000.0f, 30);
```

Comment:
Pulling the load screen drawing into here feels like an abstraction violation. But I have an idea to improve this and will do it after this PR.